lib/internal/fs/streams.js

```diff
@@ -79,6 +113,30 @@ function close(stream, err, cb) {
   }
 }
 
+function importFd(stream, options) {
+  stream.fd = null;
+  if (options.fd) {
+    // When fd is a FileHandle we can listen for 'close' events
+    if (options.fd instanceof FileHandle) {
```

Comment:
I wonder if there is a way to avoid the `instanceof` here (I believe `instanceof` is an expensive operator, but I haven't run any benchmark recently, I may be wrong here). What about checking first if `fd` is a `number`?
```suggestion
    // When fd is a raw descriptor, we must keep our fingers crossed
    // that the descriptor won't get closed, or worse, replaced with
    // another one
    // https://github.com/nodejs/node/issues/35862
    if (typeof options.fd === 'number') {
      stream.fd = options.fd;
    } else if (options.fd instanceof FileHandle) {
```

Reply:
Yeah, `instanceof` runs the constructor, good remark